_sc2data/Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UnitData.xml

```diff
@@ -8145,6 +8322,238 @@
+            <LayoutButtons Face="AP_HotSPressurizedGlands" Type="Passive" Requirements="AP_HaveHotSPressurizedGlands" Row="1" Column="2"/>
+            <LayoutButtons Face="AP_LocustSpeed" Type="Passive" Requirements="AP_HaveHotSLocustSpeed" Row="1" Column="2"/>
+            <LayoutButtons Face="AP_SwarmHostDeepBurrow" Type="AbilCmd" AbilCmd="AP_SwarmHostDeepBurrow,Execute" Row="2" Column="1"/>
+            <LayoutButtons Face="AP_SwarmHostGenerateCreep" Type="Passive" Requirements="AP_HotSHaveSwarmHostSplitB" Row="2" Column="3"/>
+        </CardLayouts>
+        <Radius value="0.8125"/>
+        <SeparationRadius value="0"/>
+        <InnerRadius value="0.5"/>
+        <SubgroupPriority value="18"/>
+        <EditorCategories value="ObjectType:Unit,ObjectFamily:Melee"/>
+        <LeaderAlias value="AP_SwarmHost"/>
+        <HotkeyAlias value="AP_SwarmHost"/>
+        <TechAliasArray value="Alias_SwarmHost"/>
+        <GlossaryAlias value="AP_SwarmHost"/>
+        <KillDisplay value="Always"/>
+        <RankDisplay value="Always"/>
+        <AIEvaluateAlias value="AP_SwarmHost"/>
+    </CUnit>
+    <CUnit id="AP_SwarmHostRooted">
+        <Name value="Unit/Name/AP_SwarmHost"/>
+        <Description value="Button/Tooltip/AP_SwarmHost"/>
+        <DeathRevealRadius value="3"/>
+        <Facing value="29.9926"/>
+        <Race value="Zerg"/>
+        <Mob value="Multiplayer"/>
+        <FlagArray index="Turnable" value="0"/>
+        <FlagArray index="PreventDestroy" value="1"/>
+        <FlagArray index="UseLineOfSight" value="1"/>
+        <FlagArray index="AIThreatGround" value="1"/>
+        <FlagArray index="AIHighPrioTarget" value="1"/>
+        <FlagArray index="AIPressForwardDisabled" value="1"/>
+        <FlagArray index="ArmySelect" value="1"/>
+        <PlaneArray index="Ground" value="1"/>
+        <Collide index="Ground" value="1"/>
+        <Collide index="Colossus" value="1"/>
+        <Collide index="ForceField" value="1"/>
+        <Collide index="Small" value="1"/>
+        <Attributes index="Armored" value="1"/>
+        <Attributes index="Biological" value="1"/>
+        <LifeStart value="120"/>
+        <LifeMax value="120"/>
+        <LifeArmor value="1"/>
+        <LifeRegenRate value="0.2734"/>
+        <LifeArmorName value="Unit/LifeArmorName/ZergGroundArmor"/>
+        <ShieldArmorName value="Unit/ShieldArmorName/ZergWildMutation"/>
+        <TurningRate value="360"/>
+        <Sight value="10"/>
+        <Food value="-3"/>
+        <CostCategory value="Army"/>
+        <CostResource index="Minerals" value="200"/>
+        <CostResource index="Vespene" value="100"/>
+        <AttackTargetPriority value="20"/>
+        <AbilArray Link="que1"/>
+        <AbilArray Link="attack"/>
+        <AbilArray Link="stop"/>
+        <AbilArray Link="AP_MorphToSwarmHostUprooted"/>
+        <AbilArray Link="AP_LocustLaunch"/>
+        <AbilArray Link="AP_SwarmHostDeepBurrow"/>
+        <AbilArray Link="AP_LocustFlyingLaunch"/>

```

Comment:
I'd expect an ability that auto-morphs the rooted variant into burrowed one. Check Refinery into automated refinery or Tech Lab into Tech Reactor